src/connection.ts

```diff
@@ -1991,38 +2401,21 @@ class Connection extends EventEmitter {
     return new TokenStreamParser(message, this.debug, handler, this.config.options);
   }
 
-  connectOnPort(port: number, multiSubnetFailover: boolean, signal: AbortSignal) {
-    const connectOpts = {
-      host: this.routingData ? this.routingData.server : this.config.server,
-      port: this.routingData ? this.routingData.port : port,
-      localAddress: this.config.options.localAddress
-    };
+  connectOnPort(port: number, multiSubnetFailover: boolean, signal: AbortSignal): Promise<Socket> {
+    return new Promise((resolve, reject) => {
+      const connectOpts = {
+        host: this.routingData ? this.routingData.server : this.config.server,
+        port: this.routingData ? this.routingData.port : port,
+        localAddress: this.config.options.localAddress
+      };
 
-    new Connector(connectOpts, signal, multiSubnetFailover).execute((err, socket) => {
-      if (err) {
-        if (err.name === 'AbortError') {
-          return;
+      new Connector(connectOpts, signal, multiSubnetFailover).execute((err, socket) => {
```

Comment:
The `Connector` class should be changed to provide an `async` API. And maybe change this from a class to just a single method?